pkg/minikube/bootstrapper/bsutil/kubeadm.go

```diff
@@ -198,3 +204,17 @@ func etcdExtraArgs(extraOpts config.ExtraOptionSlice) map[string]string {
 	}
 	return args
 }
+
+// IsKnownError replaces raw error type if the error is known.
+func IsKnownError(err error) (bool, error) {
+	// If docker runtime is not running, some docker command will crash and
+	// others will return a error message.
+	if strings.Contains(err.Error(), "github.com/docker/cli/cli/command/system.formatInfo") || strings.Contains(err.Error(), "Cannot connect to the Docker daemon") {
```

Comment:
please put the Real output in the comment so we know how u got the "github.com/docker/cli/cli/command/system.formatInfo"" as the Identifaction  for this

Reply:
Complete error output is shown below:
```
X Exiting due to K8S_INSTALL_FAILED: updating control plane: generating kubeadm cfg: getting cgroup driver: docker info --format : exit status 2
stdout:

stderr:
panic: reflect: indirection through nil pointer to embedded struct [recovered]
panic: reflect: indirection through nil pointer to embedded struct

goroutine 1 [running]:
text/template.errRecover(0xc0001d5348)
/build/docker.io-IpWXdj/docker.io-19.03.6/go/src/text/template/exec.go:166 +0x1b5
panic(0x5575eb4aed00, 0x5575eb7ecd00)
/build/docker.io-IpWXdj/docker.io-19.03.6/go/src/runtime/panic.go:522 +0x1b9
reflect.Value.FieldByIndex(0x5575eb691460, 0xc0003e07c0, 0x99, 0xc000058dc0, 0x2, 0x2, 0x0, 0x5575eb8b0380, 0x5575eb4aed00)
/build/docker.io-IpWXdj/docker.io-19.03.6/go/src/reflect/value.go:852 +0x32b
text/template.(*state).evalField(0xc0001d52c8, 0x5575eb691460, 0xc0003e07c0, 0x99, 0x7fff625d3776, 0xc, 0x5575eb86f480, 0xc000177f20, 0xc000590a50, 0x1, ...)
/build/docker.io-IpWXdj/docker.io-19.03.6/go/src/text/template/exec.go:604 +0xa59
text/template.(*state).evalFieldChain(0xc0001d52c8, 0x5575eb691460, 0xc0003e07c0, 0x99, 0x5575eb691460, 0xc0003e07c0, 0x99, 0x5575eb86f480, 0xc000177f20, 0xc000590a40, ...)
/build/docker.io-IpWXdj/docker.io-19.03.6/go/src/text/template/exec.go:554 +0x222
text/template.(*state).evalFieldNode(0xc0001d52c8, 0x5575eb691460, 0xc0003e07c0, 0x99, 0xc000177f20, 0xc000590a50, 0x1, 0x1, 0x5575eb563440, 0x5575ecaa3830, ...)
/build/docker.io-IpWXdj/docker.io-19.03.6/go/src/text/template/exec.go:518 +0x116
text/template.(*state).evalCommand(0xc0001d52c8, 0x5575eb691460, 0xc0003e07c0, 0x99, 0xc000177e30, 0x5575eb563440, 0x5575ecaa3830, 0x99, 0xc0001d50e0, 0x0, ...)
/build/docker.io-IpWXdj/docker.io-19.03.6/go/src/text/template/exec.go:456 +0x7aa
text/template.(*state).evalPipeline(0xc0001d52c8, 0x5575eb691460, 0xc0003e07c0, 0x99, 0xc00058f5c0, 0x0, 0xc00057f198, 0x0)
/build/docker.io-IpWXdj/docker.io-19.03.6/go/src/text/template/exec.go:430 +0x11e
text/template.(*state).walk(0xc0001d52c8, 0x5575eb691460, 0xc0003e07c0, 0x99, 0x5575eb86f300, 0xc0000b63f0)
/build/docker.io-IpWXdj/docker.io-19.03.6/go/src/text/template/exec.go:254 +0x49e
text/template.(*state).walk(0xc0001d52c8, 0x5575eb691460, 0xc0003e07c0, 0x99, 0x5575eb86f540, 0xc000177cb0)
/build/docker.io-IpWXdj/docker.io-19.03.6/go/src/text/template/exec.go:262 +0x145
text/template.(*Template).execute(0xc0003e0740, 0x5575eb80f2c0, 0xc0002b2e10, 0x5575eb691460, 0xc0003e07c0, 0x0, 0x0)
/build/docker.io-IpWXdj/docker.io-19.03.6/go/src/text/template/exec.go:217 +0x1ea
text/template.(*Template).Execute(...)
/build/docker.io-IpWXdj/docker.io-19.03.6/go/src/text/template/exec.go:200
github.com/docker/cli/cli/command/system.formatInfo(0x5575eb8a92e0, 0xc00030c0f0, 0x0, 0xc000590a00, 0x1, 0x1, 0xc0003e0700, 0x0, 0x0, 0x0, ...)
/build/docker.io-IpWXdj/docker.io-19.03.6/.gopath/src/github.com/docker/cli/cli/command/system/info.go:478 +0x1f6
github.com/docker/cli/cli/command/system.runInfo(0xc000593b80, 0x5575eb8a92e0, 0xc00030c0f0, 0xc0005903c0, 0x0, 0x0)
/build/docker.io-IpWXdj/docker.io-19.03.6/.gopath/src/github.com/docker/cli/cli/command/system/info.go:85 +0x344
github.com/docker/cli/cli/command/system.NewInfoCommand.func1(0xc000593b80, 0xc0002487e0, 0x0, 0x2, 0x0, 0x0)
/build/docker.io-IpWXdj/docker.io-19.03.6/.gopath/src/github.com/docker/cli/cli/command/system/info.go:52 +0x48
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).execute(0xc000593b80, 0xc000358010, 0x2, 0x2, 0xc000593b80, 0xc000358010)
/build/docker.io-IpWXdj/docker.io-19.03.6/.gopath/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:762 +0x467
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0003aaf00, 0xc000358000, 0x3, 0x3)
/build/docker.io-IpWXdj/docker.io-19.03.6/.gopath/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:852 +0x2ee
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).Execute(...)
/build/docker.io-IpWXdj/docker.io-19.03.6/.gopath/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:800
main.runDocker(0xc00030c0f0, 0x5575eb811680, 0xc0000ca010)
/build/docker.io-IpWXdj/docker.io-19.03.6/.gopath/src/github.com/docker/cli/cmd/docker/docker.go:281 +0x1d5
main.main()
/build/docker.io-IpWXdj/docker.io-19.03.6/.gopath/src/github.com/docker/cli/cmd/docker/docker.go:292 +0xf3
```
`github.com/docker/cli/cli/command/system.formatInfo` is unique to this error.